vxyowsup/tests/test_whatsapp.py

```diff
@@ -120,6 +143,17 @@ def receive(self, data):
         '''
         self.toUpper(data)
 
+    def send_ack(self, node):
+        # TODO: rather send IncomingAckProtocolEntity and test extra fields
+        ack = AckProtocolEntity(_id=node['id'], _class='message')
+        self.receive(ack.toProtocolTreeNode())
+
+    def send_receipt(self, node):
+        # msg_class=None defualt indicates "delivered"
+        # alt: msg_class='read'
+        receipt = IncomingReceiptProtocolEntity(_id=node['id'], _from=node['from'], timestamp=node['t'], type='read')
```

Comment:
I'm guessing that `_from` might be ending up as `None` here somehow. We can find out which thing ended up as none by printing out all the attributes (`for k, v in receipt.attributes.items(): print k, v`).